.github/workflows/build-test.yml

```diff
@@ -11,15 +11,15 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node-version: [16.x, 18.x, 19.x]
+        node-version: [16.x, 18.x, 20.x]
     steps:
-    - uses: actions/checkout@v3
-      with:
-        fetch-depth: 500
+    - uses: actions/checkout@v4
+    - run: git fetch origin "${{ github.event.pull_request.base.sha }}" "${{ github.event.pull_request.head.sha }}" --depth 1
```

Comment:
This line looks like it would fail on push. This workflow runs on push to `master`. We should update the references to `github.event.pull_request.base.sha` to use `github.event.push.before` as a fallback, and the references to `github.event.pull_request.head.sha` to use `github.event.push.after` as a fallback.

I see there is a pre-existing example of the `pull_request` event being referenced below though, so I guess this is already broken on push to master. We use `main` now instead of `master`; I'm guessing we forgot to update this workflow when that switch was made.

Reply:
To be honest I'm not sure how to do this fallback and absent a definite example on the internet I'm not sure how I would test it either. I can give it a shot later or we could just merge this as-is and fix the missed master->main migration in a future PR?

Reply:
Sounds great!